var/spack/repos/builtin/packages/pumi/package.py

```diff
@@ -109,15 +109,15 @@ def test(self):
         if self.spec.version <= Version("2.2.6"):
             return
         exe = "uniform"
-        options = ["../testdata/pipe.dmg", "../testdata/pipe.smb", "pipe_unif.smb"]
+        options = ["../share/testdata/pipe.dmg", "../share/testdata/pipe.smb", "pipe_unif.smb"]
         expected = "mesh pipe_unif.smb written"
         description = "testing pumi uniform mesh refinement"
         self.run_test(exe, options, expected, purpose=description, work_dir=self.prefix.bin)
 
         mpiexec = Executable(join_path(self.spec["mpi"].prefix.bin, "mpiexec")).command
```

Comment:
You don't need to pass an executable to `run_test` so you could simply set this to:

```suggestion
        mpiexec = self.spec["mpi"].prefix.bin.mpiexec
```